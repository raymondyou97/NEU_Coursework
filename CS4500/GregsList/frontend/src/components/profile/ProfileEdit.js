'use es6';

import React, { PureComponent } from 'react';

import Button from 'react-bootstrap/Button';
import FormControl from 'react-bootstrap/FormControl';
import Form from 'react-bootstrap/Form';

import '../../styles/profileEdit.css';

class ProfileEdit extends PureComponent {
  componentDidMount() {
    this.props.fetchLoggedInUserInfo();
  }

  updateUsername = e => {
    this.props.updateUsername(e.target.value);
  };

  updateFirstName = e => {
    this.props.updateFirstName(e.target.value);
  };

  updateLastName = e => {
    this.props.updateLastName(e.target.value);
  };

  updateOldPassword = e => {
    this.props.updateOldPassword(e.target.value);
  };

  updateNewPassword = e => {
    this.props.updateNewPassword(e.target.value);
  };

  updateConfirmNewPassword = e => {
    this.props.updateConfirmNewPassword(e.target.value);
  };

  passwordsMatch = () => {
    return (
      (!this.props.oldPassword &&
        !this.props.newPassword &&
        !this.props.confirmNewPassword) ||
      this.props.newPassword === this.props.confirmNewPassword
    );
  };

  renderSubmit() {
    const {
      username,
      firstName,
      lastName,
      oldPassword,
      newPassword,
      confirmNewPassword,
    } = this.props;

    const isButtonEnabled =
      username.length &&
      firstName.length &&
      lastName.length &&
      ((oldPassword.length &&
        newPassword.length &&
        confirmNewPassword.length) ||
        (!oldPassword.length &&
          !newPassword.length &&
          !confirmNewPassword.length)) &&
      this.passwordsMatch();

    return (
      <Button
        className="submit-button"
        disabled={!isButtonEnabled}
        onClick={this.props.updateUser}
        variant="primary"
      >
        Update Information
      </Button>
    );
  }

  renderInputForms() {
    return (
      <Form>
        <Form.Label className="profile-edit-label">Username</Form.Label>
        <FormControl
          className="profile-edit-input username-input"
          onChange={this.updateUsername}
          placeholder="Username"
          value={this.props.username}
        />
        <Form.Label className="profile-edit-label">First name</Form.Label>
        <FormControl
          className="profile-edit-input firstname-input"
          onChange={this.updateFirstName}
          placeholder="First name"
          value={this.props.firstName}
        />
        <Form.Label className="profile-edit-label">Last name</Form.Label>
        <FormControl
          className="profile-edit-input lastname-input"
          onChange={this.updateLastName}
          placeholder="Last name"
          value={this.props.lastName}
        />
        <Form.Label className="profile-edit-label">Old password</Form.Label>
        <FormControl
          className="profile-edit-input password-input"
          onChange={this.updateOldPassword}
          placeholder="Old password"
          type="password"
          value={this.props.oldPassword}
        />
        <Form.Label className="profile-edit-label">New password</Form.Label>
        <FormControl
          className="profile-edit-input password-input"
          onChange={this.updateNewPassword}
          placeholder="New password"
          type="password"
          value={this.props.newPassword}
        />
        <Form.Label className="profile-edit-label">
          Confirm New password
        </Form.Label>
        <FormControl
          className={`profile-edit-input confirm-input ${
            this.passwordsMatch() ? '' : 'passwords-fail'
          }`}
          onChange={this.updateConfirmNewPassword}
          placeholder="Confirm password"
          type="password"
          value={this.props.confirmNewPassword}
        />
        {this.props.nameError ? (
          <Form.Label className="profile-edit-label-error">
            Username already exists
          </Form.Label>
        ) : null}
        {this.props.oldPasswordError ? (
          <Form.Label className="profile-edit-label-error">
            Old password is incorrect
          </Form.Label>
        ) : null}
        {this.renderSubmit()}
      </Form>
    );
  }

  render() {
    return (
      <div className="profile-edit-container">
        <h1>Edit My Profile</h1>
        {this.renderInputForms()}
      </div>
    );
  }
}

export default ProfileEdit;
