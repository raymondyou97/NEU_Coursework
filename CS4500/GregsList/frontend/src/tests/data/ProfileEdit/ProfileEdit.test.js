'use es6';

import React from 'react';
import renderer from 'react-test-renderer';
import ProfileEdit from '../../../components/profile/ProfileEdit';

test('Empty snapshot renders', () => {
  const component = renderer.create(
    <ProfileEdit
      firstName={''}
      lastName={''}
      username={''}
      oldPasswordCheck={''}
      oldPassword={''}
      newPassword={''}
      confirmNewPassword={''}
      fetchLoggedInUserInfo={() => {}}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updateOldPassword={() => {}}
      updateNewPassword={() => {}}
      updateConfirmNewPassword={() => {}}
      updateUser={() => {}}
      nameError={false}
      oldPasswordError={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with all user data renders', () => {
  const component = renderer.create(
    <ProfileEdit
      firstName={'tom'}
      lastName={'sawyer'}
      username={'tsawyer'}
      oldPasswordCheck={'ts'}
      oldPassword={'ts'}
      newPassword={'tsaw'}
      confirmNewPassword={'tsaw'}
      fetchLoggedInUserInfo={() => {}}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updateOldPassword={() => {}}
      updateNewPassword={() => {}}
      updateConfirmNewPassword={() => {}}
      updateUser={() => {}}
      nameError={false}
      oldPasswordError={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data but non-matching new passwords renders', () => {
  const component = renderer.create(
    <ProfileEdit
      firstName={'tom'}
      lastName={'sawyer'}
      username={'tsawyer'}
      oldPasswordCheck={'ts'}
      oldPassword={'ts'}
      newPassword={'tsaw'}
      confirmNewPassword={'tsaw123'}
      fetchLoggedInUserInfo={() => {}}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updateOldPassword={() => {}}
      updateNewPassword={() => {}}
      updateConfirmNewPassword={() => {}}
      updateUser={() => {}}
      nameError={false}
      oldPasswordError={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data and no passwords renders', () => {
  const component = renderer.create(
    <ProfileEdit
      firstName={'tom'}
      lastName={'sawyer'}
      username={'tsawyer'}
      oldPasswordCheck={''}
      oldPassword={''}
      newPassword={''}
      confirmNewPassword={''}
      fetchLoggedInUserInfo={() => {}}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updateOldPassword={() => {}}
      updateNewPassword={() => {}}
      updateConfirmNewPassword={() => {}}
      updateUser={() => {}}
      nameError={false}
      oldPasswordError={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data and only half of password fields filled renders', () => {
  const component = renderer.create(
    <ProfileEdit
      firstName={'tom'}
      lastName={'sawyer'}
      username={'tsawyer'}
      oldPasswordCheck={'ts'}
      oldPassword={'ts'}
      newPassword={'sawy'}
      confirmNewPassword={''}
      fetchLoggedInUserInfo={() => {}}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updateOldPassword={() => {}}
      updateNewPassword={() => {}}
      updateConfirmNewPassword={() => {}}
      updateUser={() => {}}
      nameError={false}
      oldPasswordError={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
