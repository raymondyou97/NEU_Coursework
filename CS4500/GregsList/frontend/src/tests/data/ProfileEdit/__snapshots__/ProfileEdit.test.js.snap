// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty snapshot renders 1`] = `
<div
  className="profile-edit-container"
>
  <h1>
    Edit My Profile
  </h1>
  <form
    className=""
  >
    <label
      className="profile-edit-label form-label"
    >
      Username
    </label>
    <input
      className="profile-edit-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      First name
    </label>
    <input
      className="profile-edit-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      Last name
    </label>
    <input
      className="profile-edit-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      Old password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      New password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      Confirm New password
    </label>
    <input
      className="profile-edit-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Update Information
    </button>
  </form>
</div>
`;

exports[`Snapshot with all user data renders 1`] = `
<div
  className="profile-edit-container"
>
  <h1>
    Edit My Profile
  </h1>
  <form
    className=""
  >
    <label
      className="profile-edit-label form-label"
    >
      Username
    </label>
    <input
      className="profile-edit-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="tsawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      First name
    </label>
    <input
      className="profile-edit-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="tom"
    />
    <label
      className="profile-edit-label form-label"
    >
      Last name
    </label>
    <input
      className="profile-edit-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="sawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      Old password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value="ts"
    />
    <label
      className="profile-edit-label form-label"
    >
      New password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value="tsaw"
    />
    <label
      className="profile-edit-label form-label"
    >
      Confirm New password
    </label>
    <input
      className="profile-edit-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="tsaw"
    />
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Update Information
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data and no passwords renders 1`] = `
<div
  className="profile-edit-container"
>
  <h1>
    Edit My Profile
  </h1>
  <form
    className=""
  >
    <label
      className="profile-edit-label form-label"
    >
      Username
    </label>
    <input
      className="profile-edit-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="tsawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      First name
    </label>
    <input
      className="profile-edit-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="tom"
    />
    <label
      className="profile-edit-label form-label"
    >
      Last name
    </label>
    <input
      className="profile-edit-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="sawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      Old password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      New password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value=""
    />
    <label
      className="profile-edit-label form-label"
    >
      Confirm New password
    </label>
    <input
      className="profile-edit-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Update Information
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data and only half of password fields filled renders 1`] = `
<div
  className="profile-edit-container"
>
  <h1>
    Edit My Profile
  </h1>
  <form
    className=""
  >
    <label
      className="profile-edit-label form-label"
    >
      Username
    </label>
    <input
      className="profile-edit-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="tsawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      First name
    </label>
    <input
      className="profile-edit-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="tom"
    />
    <label
      className="profile-edit-label form-label"
    >
      Last name
    </label>
    <input
      className="profile-edit-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="sawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      Old password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value="ts"
    />
    <label
      className="profile-edit-label form-label"
    >
      New password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value="sawy"
    />
    <label
      className="profile-edit-label form-label"
    >
      Confirm New password
    </label>
    <input
      className="profile-edit-input confirm-input passwords-fail form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Update Information
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data but non-matching new passwords renders 1`] = `
<div
  className="profile-edit-container"
>
  <h1>
    Edit My Profile
  </h1>
  <form
    className=""
  >
    <label
      className="profile-edit-label form-label"
    >
      Username
    </label>
    <input
      className="profile-edit-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="tsawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      First name
    </label>
    <input
      className="profile-edit-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="tom"
    />
    <label
      className="profile-edit-label form-label"
    >
      Last name
    </label>
    <input
      className="profile-edit-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="sawyer"
    />
    <label
      className="profile-edit-label form-label"
    >
      Old password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value="ts"
    />
    <label
      className="profile-edit-label form-label"
    >
      New password
    </label>
    <input
      className="profile-edit-input password-input form-control"
      onChange={[Function]}
      placeholder="New password"
      type="password"
      value="tsaw"
    />
    <label
      className="profile-edit-label form-label"
    >
      Confirm New password
    </label>
    <input
      className="profile-edit-input confirm-input passwords-fail form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="tsaw123"
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Update Information
    </button>
  </form>
</div>
`;
